akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -1318,6 +1318,7 @@ final class Flow[In, Out, Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends Gr
    *
    * '''Cancels when''' downstream cancels
    */
+  @deprecated("Use recoverWithRetries instead.", "2.4.4")
```

Comment:
Just got this deprecation in some tests against akka snapshot. Should this have been on `recoverWith` methods instead? Given that the deprecation in the scaladsl is on `recoverWith` not `recover`:

https://github.com/akka/akka/blob/master/akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala#L698

Reply:
I'd say the one that was copied was a mistake. The other deprecations are on `recoverWith`. Original PR: #20137

Reply:
Here's the `Source` method showing the deprecation from 2.5.14:

https://github.com/akka/akka/blob/v2.5.14/akka-stream/src/main/scala/akka/stream/javadsl/Source.scala#L1103

I see that the Scala DSL has no such deprecation, so may be the deprecations on `recover` are misplaced in general?

Reply:
So perhaps here's the original culprit: https://github.com/akka/akka/pull/20137/files#diff-dd7989ad8864b9e69daa0ddc6619e9efR821

Reply:
Yeah, that's the original misplaced deprecation.

Reply:
Perhaps @patriknw could confirm where the deprecations should be given his original PR. I'm happy to make another change.

Reply:
Created https://github.com/akka/akka/issues/25531 to track

Reply:
And concurrently Patrik created #25530 :D